Sources/Regex/AST.swift

```diff
@@ -81,14 +80,35 @@ extension AST {
         return $0.filter(f)
       }
     }
+    func filt(_ cc: CustomCharacterClass) -> CustomCharacterClass {
```

Comment:
FWIW, I think we're going to want to get away from tree transformations and treat the AST as just a nice view of what's there.